### ISSUE TITLE
Long running connection with many DDL statements times out

### ISSUE THREAD

**Issue by lluki** [MaterializeInc/materialize#15847]
### What version of Materialize are you using?

c2affb4b90c249d970bd26ff01b871bdf7b3eaf6  (v0.30.0-dev)

### How did you install Materialize?

Materialize Cloud

### What is the issue?

I ran [this script](https://github.com/lluki/mat-meta-stresstes) that creates drops views, materialized views, insert and deletes from tables against a staging environment. It times out after a while. It happened twice now, the first timeout was discussed in this [slack thread](https://materializeinc.slack.com/archives/C0246GEHL8N/p1667220060420039).

Org ID: da15e202-1c10-4bf8-8a84-a6994358457d
Region: eu-west-1

```
Running stresstest with count=200, repetitions=6
   8s: Performed    0 of 2400 actions so far. Do=   1, Undo=   0, ObjsAlive=   1
  46s: Performed   10 of 2400 actions so far. Do=   6, Undo=   5, ObjsAlive=   1
  51s: Performed   20 of 2400 actions so far. Do=  12, Undo=   9, ObjsAlive=   3
  92s: Performed   30 of 2400 actions so far. Do=  16, Undo=  15, ObjsAlive=   1
 150s: Performed   40 of 2400 actions so far. Do=  21, Undo=  20, ObjsAlive=   1
 246s: Performed   50 of 2400 actions so far. Do=  26, Undo=  25, ObjsAlive=   1
 278s: Performed   60 of 2400 actions so far. Do=  32, Undo=  29, ObjsAlive=   3
 332s: Performed   70 of 2400 actions so far. Do=  40, Undo=  31, ObjsAlive=   9
 365s: Performed   80 of 2400 actions so far. Do=  44, Undo=  37, ObjsAlive=   7
 427s: Performed   90 of 2400 actions so far. Do=  47, Undo=  44, ObjsAlive=   3
 456s: Performed  100 of 2400 actions so far. Do=  51, Undo=  50, ObjsAlive=   1
 519s: Performed  110 of 2400 actions so far. Do=  56, Undo=  55, ObjsAlive=   1
 586s: Performed  120 of 2400 actions so far. Do=  61, Undo=  60, ObjsAlive=   1
 608s: Performed  130 of 2400 actions so far. Do=  66, Undo=  65, ObjsAlive=   1
 650s: Performed  140 of 2400 actions so far. Do=  71, Undo=  70, ObjsAlive=   1
 663s: Performed  150 of 2400 actions so far. Do=  78, Undo=  73, ObjsAlive=   5
 708s: Performed  160 of 2400 actions so far. Do=  82, Undo=  79, ObjsAlive=   3
 754s: Performed  170 of 2400 actions so far. Do=  87, Undo=  84, ObjsAlive=   3
 826s: Performed  180 of 2400 actions so far. Do=  91, Undo=  90, ObjsAlive=   1
 895s: Performed  190 of 2400 actions so far. Do=  96, Undo=  95, ObjsAlive=   1
 917s: Performed  200 of 2400 actions so far. Do= 101, Undo= 100, ObjsAlive=   1
 920s: Performed  210 of 2400 actions so far. Do= 106, Undo= 105, ObjsAlive=   1
1037s: Performed  220 of 2400 actions so far. Do= 112, Undo= 109, ObjsAlive=   3
1101s: Performed  230 of 2400 actions so far. Do= 116, Undo= 115, ObjsAlive=   1
1144s: Performed  240 of 2400 actions so far. Do= 122, Undo= 119, ObjsAlive=   3
1205s: Performed  250 of 2400 actions so far. Do= 128, Undo= 123, ObjsAlive=   5
1241s: Performed  260 of 2400 actions so far. Do= 135, Undo= 126, ObjsAlive=   9
Traceback (most recent call last):
  File "/home/lukas/mat/mat-meta-stresstest/./main.py", line 217, in <module>
    setup(conn, config)
  File "/home/lukas/mat/mat-meta-stresstest/./main.py", line 201, in setup
    act(conn)
  File "/home/lukas/mat/mat-meta-stresstest/./main.py", line 157, in __call__
    self.act.undo(conn)
  File "/home/lukas/mat/mat-meta-stresstest/./main.py", line 56, in undo
    sql(conn, "DELETE FROM input WHERE id = {}".format(self.idx))
  File "/home/lukas/mat/mat-meta-stresstest/./main.py", line 41, in sql
    sqls(conn, [cmd], fetch=fetch)
  File "/home/lukas/mat/mat-meta-stresstest/./main.py", line 38, in sqls
    raise e
  File "/home/lukas/mat/mat-meta-stresstest/./main.py", line 28, in sqls
    cur.execute(cmd)
psycopg2.errors.QueryCanceled: canceling statement due to statement timeout
```

The query that was cancelled:
`2022-11-02T10:51:33.676399: DELETE FROM input WHERE id = 21`

The complete sequence of commands:

[command_log.txt](https://github.com/MaterializeInc/materialize/files/9919240/command_log.txt)

Timestamps in the command log are in central european winter time, for correlating with CRDB load.

**Comment by benesch** [comment]
Are you sure you were running v0.30.0? That hasn’t been deployed to cloud yet, except for the canaries!

**Comment by lluki** [comment]
Yes, I deployed this manually. The exact commit id is `c2affb4b90c249d970bd26ff01b871bdf7b3eaf6`. Sorry should have mentioned this in the issue

**Comment by benesch** [comment]
Ah! Okay. I expect that this is #15842.

**Comment by chaas** [comment]
It would be great to retry this after a fix to #15842 is merged. I'll ping in this ticket once that happens.

**Comment by lluki** [comment]
Can't reproduce anymore!

My script ran in the tables resource limit after 1h unfortunately, but it did more than 15x the queries than when i first ran it, because materialize got much faster. Thus I consider it resolved.
